packages/playwright/src/worker/testInfo.ts

```diff
@@ -465,7 +465,7 @@ export class TestInfoImpl implements TestInfo {
 
   _resolveSnapshotPath(template: string | undefined, defaultTemplate: string, pathSegments: string[]) {
     const subPath = path.join(...pathSegments);
-    const parsedSubPath = path.parse(subPath);
+    const parsedSubPath = parsePathMultiExt(subPath);
```

Comment:
Why do we need these changes? Could you please provide an example of what goes wrong without the change in this line?

Reply:
as discussed offline, this is needed because `path.parse` turns `foo.snapshot.yml` into `{ name: 'foo.snapshot', ext: 'yml' }`, while our usecase needs `{ name: 'foo', ext: 'snapshot.yml' }`. This is a subtle breaking change though, because it'll also impact extensions like `.foo.txt`. So we decided to make it a special case for `.snapshot.yml` instead.